File: SistemaRural_PE/src/modelo/Medicamento.java
package modelo;

import java.time.LocalDate;

public class Medicamento {

	private String idMedicamento;
    private String nombre;
    private int stockDisponible;
    private LocalDate fechaVencimiento;
	
    public void actualizarStock(int cant) {
    	this.stockDisponible += cant;
    }
    
    public boolean estaVencido() {
    	return LocalDate.now().isAfter(fechaVencimiento);
    	
    }
    
    public String getNombre() {return nombre;}
    
    public int getStockDisponible() {return stockDisponible;}
    
}

File: Clases de Python en el Sistema Rural/Medicamento.py

from datetime import date

class Medicamento:
    def __init__(self):
        self.id_medicamento = ""
        self.nombre = ""
        self.stock_disponible = 0
        self.fecha_vencimiento = None

    def actualizar_stock(self, cant):
        self.stock_disponible += cant

    def esta_vencido(self):
        return date.today() > self.fecha_vencimiento